src/fileprovider.cs

```diff
@@ -182,11 +182,18 @@ enum NSFileProviderError : long {
 		OlderExtensionVersionRunning = -2003,
 		NewerExtensionVersionFound = -2004,
 		CannotSynchronize = -2005,
-#if MONOMAC
 		NonEvictableChildren = -2006,
 		UnsyncedEdits = -2007,
 		NonEvictable = -2008,
-#endif
+	}
+
+	[Unavailable (PlatformName.MacCatalyst)]
+	[NoiOS, Mac (12,0), NoMacCatalyst]
+	[Native, Advice ("This API is not available when using UIKit on macOS.")]
```

Comment:
And since you have `[NoMacCatalyst]` you don't need the Advice:

```suggestion
	[Native]
```